crates/bevy_sprite/src/texture_atlas.rs

```diff
@@ -1,94 +1,85 @@
+///
+/// It stores a handle to [`TextureAtlasLayout`] and the index of the current section of the atlas.
+/// The texture atlas contains various *sections* of a given texture, allowing users to have a single
+/// image file for either sprite animation or global mapping.
+/// You can change the texture [`index`](Self::index) of the atlas to animate the sprite or dsplay only a *section* of the texture
+/// for efficient rendering of related game objects.
+///
+/// Check the following examples for usage:
+/// - [`animated sprite sheet example`](https://github.com/bevyengine/bevy/blob/latest/examples/2d/sprite_sheet.rs)
+/// - [`texture atlas example`](https://github.com/bevyengine/bevy/blob/latest/examples/2d/texture_atlas.rs)
+#[derive(Component, Default, Debug, Clone, Reflect)]
 #[reflect(Component)]
-pub struct TextureAtlasSprite {
-    /// The tint color used to draw the sprite, defaulting to [`Color::WHITE`]
-    pub color: Color,
-    /// Texture index in [`TextureAtlas`]
+pub struct TextureAtlas {
+    /// Texture atlas handle
+    pub layout: Handle<TextureAtlasLayout>,
+    /// Section index in [`TextureAtlasLayout`]
     pub index: usize,
-    /// Whether to flip the sprite in the X axis
-    pub flip_x: bool,
-    /// Whether to flip the sprite in the Y axis
-    pub flip_y: bool,
-    /// An optional custom size for the sprite that will be used when rendering, instead of the size
-    /// of the sprite's image in the atlas
-    pub custom_size: Option<Vec2>,
-    /// [`Anchor`] point of the sprite in the world
-    pub anchor: Anchor,
-}
-
-impl Default for TextureAtlasSprite {
-    fn default() -> Self {
-        Self {
-            index: 0,
-            color: Color::WHITE,
-            flip_x: false,
-            flip_y: false,
-            custom_size: None,
-            anchor: Anchor::default(),
-        }
-    }
-}
-
-impl TextureAtlasSprite {
-    /// Create a new [`TextureAtlasSprite`] with a sprite index,
-    /// it should be valid in the corresponding [`TextureAtlas`]
-    pub fn new(index: usize) -> TextureAtlasSprite {
-        Self {
-            index,
-            ..Default::default()
-        }
-    }
 }
 
-impl TextureAtlas {
-    /// Create a new [`TextureAtlas`] that has a texture, but does not have
-    /// any individual sprites specified
-    pub fn new_empty(texture: Handle<Image>, dimensions: Vec2) -> Self {
+impl TextureAtlasLayout {
+    /// Create a new empty layout with custom `dimensions`
+    pub fn new_empty(dimensions: Vec2) -> Self {
         Self {
-            texture,
             size: dimensions,
             texture_handles: None,
             textures: Vec::new(),
         }
     }
 
-    /// Generate a [`TextureAtlas`] by splitting a texture into a grid where each
-    /// `tile_size` by `tile_size` grid-cell is one of the textures in the
+    /// Generate a [`TextureAtlasLayout`] as a grid where each
+    /// `tile_size` by `tile_size` grid-cell is one of *section* in the
```

Comment:
```suggestion
    /// `tile_size` by `tile_size` grid-cell is one *section* in the
```
Minor language nit here.